File: Clarity/src/main/java/org/cubexmc/clarity/ItemScope.java
package org.cubexmc.clarity;

import java.util.Arrays;
import java.util.Locale;

/** Player item containers that Clarity can scan or clean. */
public enum ItemScope {
    HAND("hand"),
    INVENTORY("inventory"),
    EQUIPMENT("equipment"),
    ENDER("ender"),
    ALL("all");

    private final String id;

    ItemScope(String id) {
        this.id = id;
    }

    public String id() {
        return id;
    }

    public static boolean isScope(String raw) {
        return parse(raw) != null;
    }

    public static ItemScope parse(String raw) {
        if (raw == null) {
            return null;
        }
        String value = raw.trim().toLowerCase(Locale.ROOT);
        return Arrays.stream(values())
                .filter(scope -> scope.id.equals(value))
                .findFirst()
                .orElse(null);
    }
}

File: Clarity/src/main/kotlin/org/cubexmc/clarity/ItemScope.kt
package org.cubexmc.clarity

import java.util.Locale

/** Player item containers that Clarity can scan or clean. */
enum class ItemScope(private val id: String) {
    HAND("hand"),
    INVENTORY("inventory"),
    EQUIPMENT("equipment"),
    ENDER("ender"),
    ALL("all"),
    ;

    fun id(): String = id

    companion object {
        @JvmStatic
        fun isScope(raw: String?): Boolean = parse(raw) != null

        @JvmStatic
        fun parse(raw: String?): ItemScope? {
            val value = raw?.trim()?.lowercase(Locale.ROOT) ?: return null
            return entries.firstOrNull { it.id == value }
        }
    }
}
